modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ListTypeDefinition/ListTypeTable.tsx

```diff
@@ -120,94 +122,196 @@ function getDataSetProps(
 
 	const addItemMenu = readOnly ? [] : [addButton];
 
-	return {
-		actionParameterName: '',
-		apiURL: `/o/headless-admin-list-type/v1.0/list-type-definitions/${pickListId}/list-type-entries`,
-		creationMenu: {
-			primaryItems: addItemMenu,
-		},
-		currentURL: window.location.pathname + window.location.search,
-		customDataRenderers: {
-			itemNameRenderer,
-		},
-		customViewsEnabled: false,
-		formName: 'fm',
-		id:
-			'com_liferay_object_web_internal_list_type_portlet_portlet_ListTypeDefinitionsPortlet-listTypeDefinitionItems',
-		itemsActions: [
-			{
-				icon: 'view',
-				id: 'addListTypeEntry',
-				label: Liferay.Language.get('view'),
-			},
-			{
-				data: {
-					id: 'delete',
-					method: 'delete',
-					permissionKey: 'delete',
+	return Liferay.FeatureFlags['LPS-168886']
```

Comment:
Hi @cgmonte,

how about checking the feature flag to add just the externalReferenceCode object into fields array, to avoid the ternary  and the code repetition.
But if there's another reason for the ternary, you can keep it, of course. Thanks =)


```
fields: [
		{
	              contentRenderer: 'itemNameRenderer',
	              expand: false,
	              fieldName: 'name',
	              label: Liferay.Language.get('name'),
	              localizeLabel: true,
	              sortable: false,
		},
		{
			expand: false,
			fieldName: 'key',
			label: Liferay.Language.get('key'),
			localizeLabel: true,
			sortable: false,
		},
		{
			...(Liferay.FeatureFlags['LPS-168886'] && {
				expand: false,
				fieldName: 'externalReferenceCode',
				label: Liferay.Language.get(
					'external-reference-code'
				),
				localizeLabel: true,
				sortable: false,
			}),
		},
	],

```

Reply:
Hi, @alinedoleron! Thanks for your suggestion! There is a problem with this code: if the feature flag is disabled, the element in the array will be a empty object that will crash FDS, so I did not incorporated this change. I know, making a ternary between two big objects is not pretty, but it works and will be removed once we remove revert the feature flag commit!

Reply:
Hi @cgmonte! You're right, my code has a failure, sorry.
I made a quick test o console.log: `obj = [{a:'one'}, ...(x===false? [{x: 'two'}] : [])]`
So, you could do something like:
```
 fields: [
		{
	              contentRenderer: 'itemNameRenderer',
	              expand: false,
	              fieldName: 'name',
	              label: Liferay.Language.get('name'),
	              localizeLabel: true,
	              sortable: false,
		},
		{
			expand: false,
			fieldName: 'key',
			label: Liferay.Language.get('key'),
			localizeLabel: true,
			sortable: false,
		},
		{
			...(Liferay.FeatureFlags['LPS-168886'] ? [{
				expand: false,
				fieldName: 'externalReferenceCode',
				label: Liferay.Language.get(
					'external-reference-code'
				),
				localizeLabel: true,
				sortable: false,
			}] : []),
		},
	],
```

Reply:
This will probably work! That's an interesting solution, but I am afraid I won't be able to added it today because I am leaving office now. Can you check with @marcelabc if this should be done? If so, I can do it tomorrow! Thanks a lot, Aline!